Modules/Core/Common/include/itkMatrix.h

```diff
@@ -194,6 +194,15 @@ class ITK_TEMPLATE_EXPORT Matrix
     m_Matrix.set_identity();
   }
 
+  /** Get an identity matrix. */
+  static Self
+  GetIdentity()
+  {
+    InternalMatrixType internalMatrix;
```

Comment:
Thanks for looking at this code so carefully, @blowekamp However, please look: `using InternalMatrixType = vnl_matrix_fixed<T, NRows, NColumns>`: https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkMatrix.h#L66

So it is a _fixed_ matrix type, without dynamic memory allocation.

Looking closer at `vnl_matrix_fixed`: its default-constructor does not do anything! So it is superfast! 😄 https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_matrix_fixed.h#L116

On the other hand, your suggested declaration `Self m;` would zero-initialize all matrix elements. So I think your suggestion might be _less_ efficient than the current PR!!! In "release mode" however, I expect your suggestion to be equally fast as the current version of this PR. 

In this particular case, return value optimization / copy elision is less relevant, because `itk::Matrix` is "trivially copyable".  (Its copy and move operations just do a bitwise copy.)

Reply:
It might be optimizing the wrong thing, but just in case ... if speed is key and space less so, we could create a `static const Matrix * const` (or similarly `static const Matrix &`) protected(?) member of `itk::Matrix` pointing to the identity matrix that is to be returned by `itk::Matrix::GetIdentity()`.  (Or use `constexpr` instead of `const` if our compilers are modern enough.)

Reply:
@Leengit Thanks for your suggestions! I guess it could be feasible to define `Matrix<T, NRows, NColumns>::GetIdentity()` as a constexpr function, once we upgrade from C++11 to C++14. 

When submitting this PR, speed was not the key thing for me. But I think it's fast enough, and I believe that it's not so easy to make it much faster now. 

My main aim for this PR was to initialize an object as _identity matrix_ more easily. Especially when using C++11 in-class member initialization. Do you agree that `GetIdentity()` serves this purpose well?

Reply:
Yes, the current implementation serves this purpose well.  But just to clarify, I think C++14 and `constexpr` are not required.  In itkMatrix.h:
```c++
static const Self & m_IdentityMatrix;

/** Get an identity matrix. */
static Self
GetIdentity()
{
    return m_IdentityMatrix;
}
```
In itkMatrx.hxx:
```c++
template <typename T, unsigned int NRows, unsigned int NColumns>
const Matrix<T, NRows, NColumns> & Matrix<T, NRows, NColumns>::m_IdentityMatrix = []() {
    InternalMatrixType internalMatrix;
    internalMatrix.set_identity();
    return Self{ internalMatrix };
}()
```
We'd have to check that all compilers are happy not de-allocating the `Self{ internalMatrix }` that is being assigned to a reference.  If does prove to be an issue then declaring `m_IdentityMatrix` _outside_ of the `class Matrix` scope as a non-reference would work.

But, this is just food for thought; the current implementation is amply good!

Reply:
@N-Dekker Thanks for pointing to the internal matrix type as vnl_matrix_fixed, I missed that. My concern was for the extra system calls in additional memory allocations, not so much the details of the most efficient identity matrix initialization.

@Leengit That is a nice optimization approach with a typical memory vs speed trade off.  As this usage is typically not in an inner loop, I don't believe this level of optimization is needed here.